src/Common/src/System/IO/PathInternal.Windows.StringBuffer.cs

```diff
@@ -15,27 +15,40 @@ internal static partial class PathInternal
         /// </summary>
         internal static bool IsExtended(StringBuffer path)
         {
-            Debug.Assert(path != null);
-            return path.StartsWith(ExtendedPathPrefix);
+            // While paths like "//?/C:/" will work, they're treated the same as "\\.\" paths.
+            // Skipping of normalization will *only* occur if back slashes ('\') are used.
+            return path.Length >= DevicePrefixLength
+                && path[0] == '\\'
+                && (path[1] == '\\' || path[1] == '?')
+                && path[2] == '?'
+                && path[3] == '\\';
```

Comment:
Yes this is only for device paths that have "extended" behavior (e.g. they skip normalization and length checks). They have to be either `\\?\` or `\??\` exactly (can't have forward seps).